src/Components/Components/src/NavigationManager.cs

```diff
@@ -308,6 +337,21 @@ protected void NotifyLocationChanged(bool isInterceptedLink)
         }
     }
 
+    /// <summary>
+    /// Triggers the <see cref="NotFound"/> event with the current URI value.
+    /// </summary>
+    protected void NotifyNotFound()
+    {
+        try
+        {
+            _notFound?.Invoke(this, new EventArgs());
+        }
+        catch (Exception ex)
+        {
+            throw new NotFoundRenderingException("An exception occurred while dispatching a NotFound event.", ex);
```

Comment:
Can we avoid the custom exception type, and let the original exception bubble up? (We normally don't create custom exception types and prefer InvalidOperationException)